src/audio/asrc/asrc.c

```diff
@@ -465,17 +311,15 @@ static int asrc_verify_params(struct processing_module *mod,
 	 * src->source/sink_rate = 0 means that source/sink rate can vary.
 	 */
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
-		if (asrc_get_source_rate(&cd->ipc_config) &&
-		    params->rate != asrc_get_source_rate(&cd->ipc_config)) {
+		if (cd->source_rate && params->rate != cd->source_rate) {
 			comp_err(dev, "asrc_verify_params(): runtime stream pcm rate %u does not match rate %u fetched from ipc.",
-				 params->rate, asrc_get_source_rate(&cd->ipc_config));
+				 params->rate, cd->source_rate);
 			return -EINVAL;
 		}
 	} else {
-		if (asrc_get_sink_rate(&cd->ipc_config) &&
-		    params->rate != asrc_get_sink_rate(&cd->ipc_config)) {
+		if (cd->sink_rate && params->rate != cd->sink_rate) {
 			comp_err(dev, "asrc_verify_params(): runtime stream pcm rate %u does not match rate %u fetched from ipc.",
-				 params->rate, asrc_get_sink_rate(&cd->ipc_config));
+				 params->rate, cd->sink_rate);
```

Comment:
no need to change line 314 - 322, it is not part of code clean or re-organize.
if you want to change, you can use a separate pr with full investigation.

Reply:
as I told you before, actually we will can use the parameters form init to replace the function call here.